cmake/toolchain/llvm/target.cmake

```diff
@@ -7,7 +7,25 @@ elseif(CONFIG_LLVM_USE_LLD)
 endif()
 
 if("${ARCH}" STREQUAL "arm")
-  set(triple arm-none-eabi)
+  if(DEFINED CONFIG_ARMV8_M_MAINLINE)
+    # ARMv8-M mainline is ARMv7-M with additional features from ARMv8-M.
+    set(triple armv8m.main-none-eabi)
+  elseif(DEFINED CONFIG_ARMV8_M_BASELINE)
+    # ARMv8-M baseline is ARMv6-M with additional features from ARMv8-M.
+    set(triple armv8m.base-none-eabi)
```

Comment:
@stephanosio are we planning to have a clang sdk-ng build, and if so, wouldn't these have *-zephyr-* tags?

Reply:
but in that case we would probably keep the `llvm` for vanilla llvm, and have a dedicated name for identifying the Zephyr llvm toolchain variant.

For now I believe the proposed solution is acceptable, and then we can adjust if need be when Zephyr SDK includes clang compiler.